src/averageAges.js

```diff
@@ -37,7 +43,17 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  let averageAge = 0;
+  let womanWith = [];
+
+  withChildren === true
+    ? womanWith = people.filter(d => d.sex === 'f'
+&& people.find(person => person.mother === d.name))
+    : womanWith = people.filter(d => d.sex === 'f');
```

Comment:
It might be better to not use ternary operator on different lines, ideally it should be written in one line. To do so, you need to create separate variables: for women (filtered out of people array) and for women with children (filtered out of women array).
Then you can use ternary operator to decide which array to use for further operations ([women] or [women with children]), depending on if withChildren argument was provided.

For example, const arrayForCalculations = withChildren ? [array of women with children] : [array of women];